src/GameManager.cpp

```diff
@@ -316,10 +279,102 @@ unsigned int GameManager::createBarricade(SDL_Renderer* gRenderer, float x, floa
 }
 
 
-void GameManager::deleteBarricade(unsigned int id) {
-    barricadeManager.erase(id);
+void GameManager::deleteBarricade(const int32_t id) {
+ barricadeManager.erase(id);
 }
 // Get a barricade by its id
-Barricade& GameManager::getBarricade(unsigned int id) {
+Barricade& GameManager::getBarricade(const int32_t id) {
     return barricadeManager.find(id)->second;
 }
+
+// Create zombie add it to manager, returns success
+int32_t GameManager::createWall(SDL_Renderer* gRenderer, const float x, const float y, const int w, const int h) {
+    const int32_t id = generateID();
+    objectManager[id] = Wall(w, h);
+    printf("%d\n", id);
+    if (!objectManager.at(id).texture.loadFromFile("assets/texture/wall.png", gRenderer)) {
+        printf("Failed to load the wall texture!\n");
+        deleteBarricade(id);
+        return -1;
+   } else {
+        objectManager.at(id).texture.setDimensions(w, h);
+    }
+    
+    objectManager.at(id).setPosition(x,y);
+    
+    return id;
+}
+
+
+void GameManager::setBoundary(SDL_Renderer* gRenderer, const float startX, const float startY, const float endX, const float endY){
+
+    int width = endX - startX + 200;
+    int height = 100;
+
+    const float x = startX - 100;
+    const float y = startY - 100;
+
+    createWall(gRenderer, x, y, width, height);
+    createWall(gRenderer, x, endY, width, height);
+ 
+    width = 100;
+    height = endY - startY + 100;
+
+    createWall(gRenderer, endX, startY, width, height);
+    createWall(gRenderer, x, startY, width, height);
+
+    const float sX = (endX + startX)/2 - BASE_WIDTH - 100;
+    const float eX = (endX + startX)/2 + BASE_WIDTH + 100;
+    const float sY = (endY + startY)/2 - BASE_HEIGHT - 100;
+    const float eY = (endY + startY)/2 + BASE_HEIGHT + 100;
+
+    width = eX - sX;
+    height = 100;
+
+    createWall(gRenderer, sX, sY, width / 2, height);
+    createWall(gRenderer, sX + (width / 4 * 3), sY, width / 4, height);
+    createWall(gRenderer, sX, eY, width / 4, height);
+    createWall(gRenderer, sX + width / 2 + 100, eY, width / 2, height);
+
+    width = 100;
+    height = eY - sY;
+
+    createWall(gRenderer, sX, sY, width, height / 2);
+    createWall(gRenderer, sX, sY + (height / 4 * 3), width, height / 4);
+    createWall(gRenderer, eX, sY, width, height / 1.5);
+    createWall(gRenderer, eX, sY + (height / 4 * 3), width, height / 4);
+}
+
+bool GameManager::createZombieWave(SDL_Renderer* gRenderer, const int n){
+    std::vector<Point> spawnPoints;
+    
+    spawnPoints.push_back(Point(-900, -900));
+    spawnPoints.push_back(Point(1900, -900));
+    spawnPoints.push_back(Point(2900, -900));
+    spawnPoints.push_back(Point(2900, 2900));
+    spawnPoints.push_back(Point(1900, 2900));
+    spawnPoints.push_back(Point(-900, 2900));
+    
+    if(zombieManager.size() >= spawnPoints.size() * 5) {
+        unsigned int count = 0;
+        std::vector<int32_t> ids;
+        for(const auto z : zombieManager) {
```

Comment:
Any reason zombies are being copied inside the loop?
Should be capturing z by reference